core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/factory.go

```diff
@@ -0,0 +1,16 @@
+package logprovider
+
+import (
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+func New(lggr logger.Logger, poller logpoller.LogPoller, utilsABI abi.ABI) (LogEventProvider, LogRecoverer) {
+	filterStore := NewUpkeepFilterStore()
+	packer := NewLogEventsPacker(utilsABI)
+	provider := NewLogProvider(lggr, poller, packer, filterStore, nil)
```

Comment:
🏄 If `filterStore` is only required for the `NewLogProvider` function, does it make sense to instantiate it within `NewLogProvider`? Also we could just pass the utilsABI as a parameter to `NewLogProvider`, and instantiate the packer in there too?

Reply:
Defined outside as the filter store is used by both the provider and recoverer